romancal/datamodels/tests/test_models.py

```diff
@@ -1,16 +1,190 @@
-import asdf
+import pytest
 
-from romancal.datamodels import RomanDataModel
+import asdf
+import numpy as np
+from astropy.time import Time
 
+from romancal import datamodels
+from stdatamodels.validate import ValidationWarning
 
-def test_model_schemas():
-    def iter_subclasses(model_class):
+# Helper class to iterate over model subclasses
+def iter_subclasses(model_class, include_base_model=True):
+    if include_base_model:
         yield model_class
-        for sub_class in model_class.__subclasses__():
-            yield from iter_subclasses(sub_class)
+    for sub_class in model_class.__subclasses__():
+        yield from iter_subclasses(sub_class)
 
-    for model_class in iter_subclasses(RomanDataModel):
+def test_model_schemas(tmp_path):
+    for model_class in iter_subclasses(datamodels.RomanDataModel):
         try:
             asdf.schema.load_schema(model_class.schema_url)
         except Exception:
-            assert False, f"Unable to load schema URI '{model_class.schema_url}' for model class {model_class.__name__}"
+            assert False, f"Unable to load schema URI '{model_class.schema_url}' for model " \
+                          f"class {model_class.__name__}"
+
+
+# Helper function to test required keywords
+# meta = dictionary of required key/value pairs to ensure each are required
+# model_class = the class of the model to be tested
+def confirm_required_keywords(tmp_path, meta, model_class):
```

Comment:
I had a chat with @nden and @jdavies-st about this and our general advice is not to include this kind of check in the romancal tests.  This helper is testing a) that the schema has certain required attributes and b) that stdatamodels is properly enforcing that requirement.  We have tests in stdatamodels itself that cover its ability to enforce a schema, so b) is redundant.

For a), the question is, what kind of bug or issue would this catch?  We would have to update the test for a new required attribute but forget to add it to the schema.  Or we'd see a failure if we added a new required attribute to the schema but forgot to maintain the test.  But in both cases these are issues introduced by maintaining two parallel lists of required attributes.  In other words, this test creates a new opportunity for bugs and then guards against them.  IMO there's little enough net value that it would be better not to have the test.

Reply:
Agree.

Also, I would advise against having try/except blocks in tests, as that exception handling will mask the actual error you're seeing in the test.  Rely on `pytest` assertion rewriting to give a useful message, and if you're expecting an exception use the `with pytest.raises()` context manager to find it.

Reply:
I am guessing that I should keep the failed enum case as a known failed test?

Reply:
If so, these changes are implemented and ready for review.